### PR TITLE
Bluetooth: Mesh: Align light dimmer sample with CDP1 changes

### PR DESCRIPTION
**Pull request by PavelVPV**
Align light dimmer sample with CDP1 change.

### CODE REVIEW

File: samples/bluetooth/mesh/light_dimmer/prj.conf [hunk 1]
```diff
@@ -49,7 +49,11 @@ CONFIG_BT_MESH_GATT_PROXY=y
 CONFIG_BT_MESH_PROXY_USE_DEVICE_NAME=y
 CONFIG_BT_MESH_DK_PROV=y
 CONFIG_BT_MESH_NLC_PERF_CONF=y
-CONFIG_BT_MESH_MODEL_EXTENSION_LIST_SIZE=1
+
+# Disable model extension API and set list size to 0 as the models in this sample don't have any
+# relation between each other.
+CONFIG_BT_MESH_MODEL_EXTENSIONS=n
+CONFIG_BT_MESH_MODEL_EXTENSION_LIST_SIZE=0
```
Comment (Balaklaka) [inline samples/bluetooth/mesh/light_dimmer/prj.conf:56]: This is not needed with changes done in the other PR.

Comment (PavelVPV) [reply samples/bluetooth/mesh/light_dimmer/prj.conf:56]: The list size is non-zero by default. The extension API is enabled by NLC profile Kconfig option:
https://github.com/nrfconnect/sdk-nrf/blob/e9f512477c9243727e2d391d05f7d4acee124008/subsys/bluetooth/mesh/Kconfig#L54-L55

**Review by Andrewpini — APPROVED** [review]

File: samples/bluetooth/mesh/light_dimmer/prj.conf (hunk 1 above)
Comment (Balaklaka) [reply samples/bluetooth/mesh/light_dimmer/prj.conf:56]: What I meant was that adding "https://github.com/nrfconnect/sdk-zephyr/pull/1536#pullrequestreview-1903761572" will the CONFIG_BT_MESH_MODEL_EXTENSION_LIST_SIZE be default = 0, and then not needed.

**Review by Balaklaka — APPROVED** [review]

File: samples/bluetooth/mesh/light_dimmer/prj.conf (hunk 1 above)
Comment (PavelVPV) [reply samples/bluetooth/mesh/light_dimmer/prj.conf:56]: The reason why the default value is not 0 is that then any attempt of adding a new entry to the list will fail. But perhaps this should be changed. Anyway, it should be done in upstream first.

**Comment by jfischer-no** [comment]
@PavelVPV  When you push a new sdk-zephyr revision, CI will probably be red because of https://github.com/nrfconnect/sdk-nrf/pull/14136

File: west.yml [hunk 2]
```diff
@@ -61,7 +61,7 @@ manifest:
     # https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/modules.html
     - name: zephyr
       repo-path: sdk-zephyr
-      revision: f3f276dd2ecad981a2abf1d87c3d3d0b2e9d5373
+      revision: 7735bc6d171ab2fe6590ab47171f5decdbe2ddfd
```
Comment (jfischer-no) [inline west.yml:64]: Please update to
e4827fba2254402b99a7be249974de4e0acbcba1
https://github.com/nrfconnect/sdk-zephyr/commit/e4827fba2254402b99a7be249974de4e0acbcba1

Comment (PavelVPV) [reply west.yml]: Done

**Review by jfischer-no — APPROVED** [review]

**Comment by krish2718** [comment]
Regarding Wi-Fi sample flash overflow, works fine with e21b491125b but pulling in the Zephyr update broke this, the best we can do for now is below till we investigate what has caused the memory hike. You can cherry pick https://github.com/krish2718/sdk-nrf/commit/20b772de1b0cd48453b584a28403ceb114b2b3cb  to fix the issue in your PR

**Comment by jfischer-no** [comment]
@PavelVPV Please rebase and drop manifest changes.

**Pull request merged**